docs(react-native): Expo cross-platform social sign-in with complete example

massimodeluisa opened the pull request:
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Adds a new tutorial to the Supabase docs ([/docs/guides/getting-started/tutorials/with-expo-react-native-social-auth](https://github.com/massimodeluisa/supabase/tree/feature/expo-social-auth/apps/docs/content/guides/getting-started/tutorials/with-expo-react-native-social-auth.mdx)) on building a social auth app with Expo React Native. Includes a complete example Expo project using the latest Expo SDK, Expo Router, and Supabase JS client for database queries, authentication, and social sign-in flows with Apple and Google providers.

## What is the current behavior?

- https://github.com/supabase/supabase/issues/30348
- https://github.com/supabase/supabase/issues/37881
- https://github.com/supabase/supabase/issues/35223

## What is the new behavior?

<img width="1728" height="1080" alt="supabase-expo-social-auth-login" src="https://github.com/user-attachments/assets/654df5a7-e636-4941-91cd-49c3080703f7" />
<img width="1728" height="1080" alt="supabase-expo-social-auth-tabs" src="https://github.com/user-attachments/assets/3e6b4fc9-927a-485a-916d-1f56d8ac3ee7" />

## Additional context

This PR adds a new tutorial to the Supabase docs to address community-reported issues, including:
- A clear integration of Expo Router for navigation with Supabase Auth for protected routes, session management, and splash screen handling.
- Implementation details for Google and Apple sign-in buttons, covering limitations of libraries like [Invertase's React Native Apple Authentication library](https://github.com/invertase/react-native-apple-authentication) (mobile), [react-apple-signin-auth](https://react-apple-signin-auth.ahmedtokyo.com/) (web), [Expo's Apple Authentication library](https://docs.expo.dev/versions/latest/sdk/apple-authentication/) (iOS), [GN Google Sign In Premium](https://react-native-google-signin.github.io/docs/install#sponsor-only-version) (mobile) and [@react-oauth/google](https://github.com/MomenSherif/react-oauth#googlelogin) (web)
- oauth/google (web Google auth), and Supabase's signInWithOAuth (mobile OAuth).
- Secure storage options (AsyncStorage for web, Expo SecureStore for mobile with encryption).
- Full auth flows (including profile fetching and logout).
- Setup guidance and testing tips.

It also adds a the complete Expo project (using latest Expo SDK, Expo Router, and Supabase JS for DB queries, auth, and Apple/Google social sign-in) at [/examples/auth/expo-social-auth](https://github.com/massimodeluisa/supabase/tree/feature/expo-social-auth/examples/auth/expo-social-auth) .

massimodeluisa committed:
docs(react-native): create the basic expo project

massimodeluisa committed:
docs(react-native): cross-platform Apple social sign-in

massimodeluisa committed:
docs(react-native): cross-platform Google social sign-in

massimodeluisa committed:
docs(react-native): fix typos

massimodeluisa committed:
Merge branch 'master' into feature/expo-social-auth

saltcod commented on apps/studio/components/interfaces/Connect/Connect.constants.ts:
```diff
@@ -256,6 +256,20 @@ export const MOBILES: ConnectionType[] = [
     ],
   },
+  {
+    key: 'exporeactnativesocialauth',
```
I think I'd prefer to leave this out of the Studio connect modal. That's mean to be a quick connect sort of thing. 

This link though could definitely mention the social auth approach
https://share.cleanshot.com/8Pqd16hf

massimodeluisa commented on apps/docs/content/guides/getting-started.mdx:
```diff
@@ -305,6 +305,14 @@ hideToc: true
       hasLightIcon: true
     },
+    {
+      title: 'Expo React Native Social Auth',
```
> I'm generally wondering if we should:
>
> - revamp the current Expo guide
> - add this as a section
> Instead of having two separate guides with a bunch of overlap

@saltcod it would be great to have a single documentation entry point.

My only concern when I wrote this separately was to avoid including too much information and steps in a single tutorial page. In my opinion, until someone creates a universal React Native (Expo) authentication library (compatible with Supabase Auth), it would probably be better to keep the authentication guide separate from other Supabase integrations. 

But this is just my opinion.

----

> @saltcod I have a separate task to look at the existing Expo content, but I am not sure when. In general, I am trying to consolidate some of the overlapping tutorials. But either way, @massimodeluisa , what you've added won't be wasted and it's appreciated!

@ChrisChinchilla awesome!

Are you planning to integrate the example I created here? It seems many devs on X are asking for this...

Plus, two devs messaged me privately to request an example of infinite scroll fetching . I provided them with a combined use of [Shopify's FlashList](https://github.com/Shopify/flash-list) and [@tanstack/react-query](https://tanstack.com/query). It would be amazing to include examples like this as well.

ChrisChinchilla replied:
@massimodeluisa Probaly yes, just how is what I need to think about. Will get back to you asap.

massimodeluisa committed:
Merge branch 'supabase:master' into feature/expo-social-auth

massimodeluisa committed:
docs(react-native): remove wrong entry in the `Connection` component

ChrisChinchilla committed:
Correct typos

ChrisChinchilla committed:
Merge branch 'master' into feature/expo-social-auth

# Conflicts:
#	pnpm-lock.yaml

ChrisChinchilla committed:
Prettier

ChrisChinchilla committed:
Draft

ChrisChinchilla reviewed (commented):
@massimodeluisa I am still working through this, but some intiial questions…

ChrisChinchilla committed:
Draft

massimodeluisa committed:
docs(react-native): use kebab-case file naming convention in Expo social auth example

massimodeluisa committed:
docs(react-native): update the packages of the Expo social auth example

massimodeluisa committed:
docs(react-native): use kebab-case file naming convention in Expo guide

- use kebab-case file naming convention in Expo guide
- add trailing semicolon to align with the standard Expo template conventions

ChrisChinchilla committed:
Fix

ChrisChinchilla commented on apps/docs/content/guides/getting-started/tutorials/with-expo-react-native-social-auth.mdx:
````diff
@@ -0,0 +1,1386 @@
+npx expo start --tunnel
+```
+{/* TODO: HERE */}
+And add it to the "Allowed Callback URLs" list in your Supabase dashboard Apple provider configuration.
````
@massimodeluisa Another little one… You mention the dashboard, but this field doesn't exist, and the auth team says it's only configurable via the CLI?

https://github.com/supabase/cli/blob/develop/pkg/config/templates/config.toml#L271-L272

Or did you mean something else?

ChrisChinchilla committed:
Draft

ChrisChinchilla committed:
Changes

ChrisChinchilla committed:
Correct log message

ChrisChinchilla committed:
Merge branch 'master' into feature/expo-social-auth

ChrisChinchilla reviewed (approved):
Thanks so much @massimodeluisa !
